File: DaiPhucVinh/Services/MainServices/CategoryList/CategoryListService.cs
using DaiPhucVinh.Server.Data.DaiPhucVinh;
using DaiPhucVinh.Services.Database;
using DaiPhucVinh.Services.Framework;
using DaiPhucVinh.Shared.CategoryList;
using DaiPhucVinh.Shared.Common;
using System;
using System.Collections.Generic;
using System.Data.Entity;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DaiPhucVinh.Services.MainServices.CategoryList
{
    public interface ICategoryListService
    {
        Task<BaseResponse<bool>> CreateCategoryList(CategoryListRequest request);
        Task<BaseResponse<bool>> UpdateCategoryList(CategoryListRequest request);
        Task<BaseResponse<bool>> DeleteCategoryList(CategoryListRequest request);
    }
    public class CategoryListService : ICategoryListService
    {
        private readonly DataContext _datacontext;
        private readonly ILogService _logService;

        public CategoryListService(DataContext datacontext, ILogService logService)
        {
            _datacontext = datacontext;
            _logService = logService;
        }
        
        public async Task<BaseResponse<bool>> CreateCategoryList(CategoryListRequest request)
        {
            var result = new BaseResponse<bool> { };
            try
            {

                var categoryList = new EN_CategoryList
                {
                    CategoryName = request.CategoryName,
                    UserId = request.UserId,
                    Status = request.Status,
                };
                _datacontext.EN_CategoryList.Add(categoryList);
                await _datacontext.SaveChangesAsync();
                result.Success = true;
            }
            catch (Exception ex)
            {
                result.Message = ex.ToString();
                _logService.InsertLog(ex);
            }
            return result;
        }

        public async Task<BaseResponse<bool>> UpdateCategoryList(CategoryListRequest request)
        {
            var result = new BaseResponse<bool> { };
            try
            {
                if (request.CategoryId > 0)
                {
                    var categoryList = await _datacontext.EN_CategoryList.FindAsync(request.CategoryId);
                    {
                        categoryList.CategoryName = request.CategoryName;
                    };
                }
                await _datacontext.SaveChangesAsync();
                result.Success = true;
            }
            catch (Exception ex)
            {
                result.Message = ex.ToString();
                _logService.InsertLog(ex);
            }
            return result;
        }
        public async Task<BaseResponse<bool>> DeleteCategoryList(CategoryListRequest request)
        {
            var result = new BaseResponse<bool> { };
            try
            {
                var categoryList = await _datacontext.EN_CategoryList.FirstOrDefaultAsync(x => x.CategoryId == request.CategoryId);
                _datacontext.EN_CategoryList.Remove(categoryList);
                await _datacontext.SaveChangesAsync();
                result.Success = true;
            }
            catch (Exception ex)
            {
                result.Message = ex.ToString();
                _logService.InsertLog(ex);
            }
            return result;
        }
    }
}

File: DaiPhucVinh/eattingnowapp/src/api/CategoryList/categoryListService.js
import { Proxy } from "../Proxy";
import { ServerEndPoint } from "../ServerEndPoint";
import { BaseResponse } from "../BaseResponse";
const ServiceEndPoint = ServerEndPoint + "/categorylist";
const API = {
  CreateCategoryList: "/CreateCategoryList",
  DeleteCategoryList: "/DeleteCategoryList",
  UpdateCategoryList: "/UpdateCategoryList",
};

const CreateCategoryList = async (request) => {
  let result = new BaseResponse(false, "", null);
  try {
    let response = await Proxy(
      "post",
      ServiceEndPoint + API.CreateCategoryList,
      request,
      true
    );
    if (response.data.Success) {
      result.item = response.data.Item;
      result.success = true;
    }
  } catch (e) {
    result.message = e.toString();
  }
  return result;
};
const DeleteCategoryList = async (e) => {
  let result = new BaseResponse(false, "", null);
  try {
    let response = await Proxy(
      "post",
      ServiceEndPoint + API.DeleteCategoryList,
      {
        CategoryId: e,
      },
      true
    );
    if (response.data.Success) {
      result.success = true;
    }
  } catch (e) {
    result.message = e.toString();
  }
  return result;
};
const UpdateCategoryList = async (request) => {
  let result = new BaseResponse(false, "", null);
  try {
    let response = await Proxy(
      "post",
      ServiceEndPoint + API.UpdateCategoryList,
      request,
      true
    );
    if (response.data.Success) {
      result.success = true;
    }
  } catch (e) {
    result.message = e.toString();
  }
  return result;
};
export { CreateCategoryList, DeleteCategoryList, UpdateCategoryList };
